core/divelist.c

```diff
@@ -805,6 +805,10 @@ static int comp_dives(const struct dive *a, const struct dive *b)
 		if (trip_date(a->divetrip) > trip_date(b->divetrip))
 			return 1;
 	}
+	if (a->number < b->number)
+		return -1;
+	if (a->number > b->number)
+		return 1;
```

Comment:
Honest question... should that also work if one of them has number 0 (i.e., isn't numbered)?

Reply:
Dives without numbers are sorted before dives with numbers (that means the other way around by default, since we default to descending).

We could reverse that (no numbers after numbers) but I don't think that's a use-case that we should spend too much thought on. Even the whole same time, but different number case appears a bit obscure to me.

Reply:
The same time case was actually a failure to import CVS dive times correctly... but this does seem the right thing to do... and I agree with your analysis.